aiidalab/app.py

```diff
@@ -496,6 +496,10 @@ def refresh(self):
             with self.hold_trait_notifications():
                 self.available_versions = list(self._available_versions())
                 self.installed_version = self._installed_version()
+                if (
+                    not self.is_installed()
+                ):  # Do not check compatibility if the app is not installed.
+                    return
```

Comment:
Instead of returning the function here, we should move the compatibility check into the `else:` block (line 510/514).

Reply:
The issue is that the `self._is_compatible(self.installed_version)` fails if `self.installe_version` is equal to "not installed"

Reply:
The other thing one can do is to add this check directly into the `self._is_compatible` function. What do you think?

Reply:
Yes, I think that was the original intention anyways. I think all we might need to do is remove the "raise" statement in https://github.com/aiidalab/aiidalab/blob/19f35375ffecec90ac969c7cebfef497b9d48f2f/aiidalab/app.py#L476 ? Looks like it was accidentally not removed.

Reply:
I can try that.

Reply:
yep, removing the `raise` seem to be enought.